### DISCUSSION TITLE
Dashes in CSS class names with Gatsby v3 (css-loader v5)

### DISCUSSION THREAD

**Discussion by alejandraarri** [gatsbyjs/gatsby#30499] [Help]
In Gatsby v2 you could enable this syntax `className={styles["block__element--modifier"]}`.

However Gatsby v3 is using css-loader v5 which no longer accepts the `camelCase: false` field in options. Instead it has the field `exportLocalsConvention` which can take the value `asIs`, but when enabling `namedExport` the compiler throws this error:

```
The "modules.namedExport" option requires the "modules.exportLocalsConvention" option to be "camelCaseOnly" or "dashesOnly".
```

I've tried to configure this in the `gatsby-config.js` file:

```js
{
	resolve: 'gatsby-plugin-sass',
	options: {
		implementation: require("node-sass"),
		cssLoaderOptions: {
			modules: {
				namedExport: false,
				exportLocalsConvention: 'asIs'
			},
		},
	},
},
```
But this doesn't make the CSS class names accessible in JSX and gives the warning:

```
warn Attempted import error: 'block' is not exported from './styles.module.css' (imported as 'styles'). 
```

It looks like `namedExport` must be set to true in order to access the compiled class names in JSX however then `css-loader` will only accept camelCaseOnly or dashesOnly for exportLocalsConvention.

I'm not sure how to enable the same functionality as Gatsby v2 in v3. I'd like to be able to continue using the `className={styles["block__element--modifier"]}` syntax because it makes identifying CSS classes easy and also just keeps things consistent between CSS and JSX (also I'd prefer not having to rewrite a bunch of my code).

This was originally posted in StackOverflow https://stackoverflow.com/questions/66633882/dashes-in-css-class-names-with-gatsby-v3-css-loader-v5. I just copy-pasted his issue because we are seeing the same error.

**Comment by vladar** [comment]
@alejandraarri This seems to be slightly tangential to Gatsby. I suggest opening an issue/question in the [css-loader repo](https://github.com/webpack-contrib/css-loader). Then we'll see what they suggest and what can be done in Gatsby to support this.

**Reply by alejandraarri** [reply]
Ok, we will do that. Thank you for your quick response!
